base/shell/cmd/cmd.c

```diff
@@ -2333,6 +2333,7 @@ static VOID Cleanup(VOID)
         ParseCommandLine(_T("\\cmdexit.bat"));
     }
 
+#ifdef _DEBUG_MEM
```

Comment:
As usual, github's interface cannot allow adding comments/suggestions in code parts that are outside of the +-3 lines around the current diff'ed lines, so...

My suggestion here: since the console mode state is something that exists across different processes living in said console, the two SetConsoleMode calls below are necessary.
So you need to end the first `_DEBUG_MEM` block with an `#endif _DEBUG_MEM` just after the `RemoveBreakHandler();`, and then restart an `#ifdef _DEBUG_MEM` just before the `DeleteCriticalSection()` call.